From da78c4a70005910503ce238b7e41b56aa9a66be6 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Mon, 7 Sep 2020 09:09:57 +0800
Subject: [PATCH] thermal: rockchip: fix up the code to temp maybe overflow

For linear calculation, code needs to be converted to Int,
otherwise the calculated value may overflow at ultra-low temperature.

Change-Id: I64c45b1f5ebc86da853180ca7bb3cb83234f2a64
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 drivers/thermal/rockchip_thermal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/thermal/rockchip_thermal.c b/drivers/thermal/rockchip_thermal.c
index e48c0b4c55f6..acb6c0b7e117 100644
--- a/drivers/thermal/rockchip_thermal.c
+++ b/drivers/thermal/rockchip_thermal.c
@@ -623,7 +623,7 @@ static int rk_tsadcv2_code_to_temp(const struct chip_tsadc_table *table,
 	unsigned long denom;
 
 	if (table->kNum) {
-		*temp = ((code - table->bNum) * 10000 / table->kNum) * 100;
+		*temp = (((int)code - table->bNum) * 10000 / table->kNum) * 100;
 		if (*temp < MIN_TEMP || *temp > MAX_TEMP)
 			return -EAGAIN;
 		return 0;
-- 
2.35.3

